src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -538,17 +538,21 @@ export const selectBalanceChartCryptoBalancesByAccountIdAboveThreshold =
       ).reduce((acc, account) => {
         Object.values(account?.stakingDataByValidatorId ?? {}).forEach(
           stakingDataByAccountSpecifier => {
-            Object.values(stakingDataByAccountSpecifier).forEach(stakingData => {
-              const { delegations, redelegations, undelegations } = stakingData
-              const redelegationEntries = redelegations.flatMap(
-                redelegation => redelegation.entries,
-              )
-              const combined = [...delegations, ...redelegationEntries, ...undelegations]
-              combined.forEach(entry => {
-                const { assetId, amount } = entry
-                acc[assetId] = bnOrZero(acc[assetId]).plus(amount).toString()
-              })
-            })
+            Object.entries(stakingDataByAccountSpecifier).forEach(
+              ([stakingAccountId, stakingData]) => {
+                // if passed an accountId filter, only aggregate for the given accountId
+                if (accountId && stakingAccountId !== accountId) return
```

Comment:
this is actually the diff - the rest is formatting

we previously weren't actually applying the accountId filter - and were aggregating all staking (delegation/undelegation/redelegation) balances across all accountIds.

this correctly applies the filter and fixes the chart